projects/plugins/jetpack/modules/widgets/contact-info.php

```diff
@@ -31,6 +31,7 @@ public function __construct() {
 				'classname'                   => 'widget_contact_info',
 				'description'                 => __( 'Display a map with your location, hours, and contact information.', 'jetpack' ),
 				'customize_selective_refresh' => true,
+				'show_instance_in_rest'       => true,
 			);
```

Comment:
Not sure if you left it out on purpose, but the hook to remove it from the Legacy block needs to be added as well. Made testing easier though : )

Reply:
Thanks! :) 

Yes, I left in out on purpose - as I wanted to hear your opinions on also leaving the Contact Info & Map widget there for the users to choose (in case they prefer it over Contact Info block).

Reply:
My understanding was that we were getting rid of the redundancies in the widget area. I think it is a little confusing to users to have two or three options for the same things. This seemed like a 1:1 transform, does it differ greatly between the widget and block option?

I am not completely sure though, it could be best to get @simison and @yansern opinion.

Reply:
>  I think it is a little confusing to users to have two or three options for the same things.

I see what you mean by looking at this from the perspective of user's confusion.

> This seemed like a 1:1 transform, does it differ greatly between the widget and block option?

The main difference I can see is the map solution being used: Google Maps (Contact Info & Map widget) vs MapBox (when the user transforms the widget into blocks).

I personally prefer MapBox, but I can see some users that would like to use Google Maps instead.

If we want to remove the redundancy, then yes, I think it would be best to remove the legacy widget.

Reply:
I see that the only way to use Google Maps is through shortcodes.
`[googlemaps https://maps/url/here]`

1. Are we able to construct a Google Maps embed URL with long & lat param?
2. Is it alright if it's a shortcode block?

Reference: https://wordpress.com/support/google-maps/#embedding-a-google-map

### How about creating a block for this?

Also, another aspect we can consider, is to create a Contact Info & Maps block **as part of our Gutenberg training**. Since the widget code lives in Jetpack, why not create the same block in Jetpack?

Then, we can come back to this transformation exercise and have it do a 1:1 transformation.

Reply:
> Are we able to construct a Google Maps embed URL with long & lat param?

For our use case, I think we could construct the Google Map embed URL using information from the physical address that the user provides, e.g. as we can see in [this example code](https://developers.google.com/maps/documentation/embed/get-started):

`https://www.google.com/maps/embed/v1/place?key=API_KEY&q=Space+Needle,Seattle+WA`

> Is it alright if it's a shortcode block?

I would personally rather moved away from shortcode if possible and voted for a block solution.

We could also try to utilize the existing Map block and use [forward geocoding](https://docs.mapbox.com/api/search/geocoding/#forward-geocoding) to get the necessary coordinates for the map.

Reply:
> The main difference I can see is the map solution being used: Google Maps (Contact Info & Map widget) vs MapBox (when the user transforms the widget into blocks).

This is indeed a big difference. Apart from the look of each service, the requirements, pricing, and API setup are also different. 

Switching from one to the next is not straightforward on self-hosted sites, where folks will need to have an API setup step (that's not necessary on WordPress.com where we handle both services for them; there switching from one service to the next is indeed straightforward).

I wonder if we should take that into account in general; if someone has a Contact Info Widget already set up, with a map, then we know what their requirements are. We could then check if they also have a Mapbox API setup on that site, and offer the transform if we know that's possible for them.

What do you think?

> I would personally rather moved away from shortcode if possible and voted for a block solution.

+1.

Reply:
> This is indeed a big difference. Apart from the look of each service, the requirements, pricing, and API setup are also different.

> Switching from one to the next is not straightforward on self-hosted sites [...]

That's a good point. 

> [...] if someone has a Contact Info Widget already set up, with a map, then we know what their requirements are. We could then check if they also have a Mapbox API setup on that site, and offer the transform if we know that's possible for them.

> What do you think?

I think we could create a new block that would be an equivalent to the existing widget - and rely on Google Maps initially. This would let us move faster with the retirement of legacy widgets.

Then, we could add a dropdown to the new block settings where the user would be able to choose whether they want to use Google Maps or Mapbox.

This would provide flexibility when it comes to block transform. As you suggested, we could then also check whether the user has the Mapbox API set up already and if yes, we could transform the legacy widget to the new block with Mapbox map.

Even if we decide not to implement Mapbox into the new block (and leave Google Maps support there only), I would consider it as a good outcome - as the users would still have the option to use the Mapbox block separately. I think this would also make the block transform more straightforward - as we would be transforming widget into block 1:1 (the same fields would be available in the block as they currently are in the widget).

In other words, as a first iteration we could have a block that has the same functionality as the current legacy widget. Then we could build on top of that and add extra features, e.g. Mapbox support.

How does that sound? 🙂

Reply:
> In other words, as a first iteration we could have a block that has the same functionality as the current legacy widget. Then we could build on top of that and add extra features, e.g. Mapbox support.

I agree with starting simple. Showing the same form as what the legacy widget did, and use Google Maps, would be the path to least resistance. I imagine literally extract bits & pieces and put them back using Block's library.

Reply:
It seems that there's a Contact Info block on Jetpack. It also creates a Google Maps.
Perhaps the work needed is to modify this block and give user the option to embed the Google Maps under.
https://github.com/Automattic/jetpack/tree/master/projects/plugins/jetpack/extensions/blocks/contact-info

Reply:
> How does that sound? 🙂

Yeah, we could give that a try. I would, however, suggest that you post on a P2 about this as there are some product discussions to be had about this. See p7fD6U-2XB-p2 for some background.

> It also creates a Google Maps

It's just a link though, so nothing super fancy. But I agree we should start there with that block, maybe by leveraging the existing map block as a new possible inner block for that block.

Reply:
I have just publish a P2 post to gather more insights so we can then decide on the best approach moving forward: p7fD6U-4KX-p2.

Reply:
Discussion continues in p7fD6U-4KX-p2